rts/lra/client/src/main/java/io/narayana/lra/client/internal/NarayanaLRAClient.java

```diff
@@ -746,20 +747,20 @@ public URI enlistCompensator(URI uri, Long timelimit, String linkHeader, StringB
                     .get(JOIN_TIMEOUT, TimeUnit.SECONDS);
 
             String responseEntity = response.hasEntity() ? response.readEntity(String.class) : "";
+            // remove it and create tests for PRECONDITION_FAILED and NOT_FOUND
             if (response.getStatus() == Response.Status.PRECONDITION_FAILED.getStatusCode()) {
-                String logMsg = LRALogger.i18nLogger.error_tooLateToJoin(lraId, responseEntity);
+                String logMsg = LRALogger.i18nLogger.error_tooLateToJoin(String.valueOf(lraId), responseEntity);
```

Comment:
I expect this has a reason to need to be String.valueOf - I am not necessarily asking for that to be explained in the code but I am curious why it needed to changed?

Reply:
Actually, I presume it has something to do with how a URL is interpreted in https://github.com/jbosstm/narayana/pull/2331/files#diff-e4c877e90cd6988c76f4b37dbd005f96444a75919262742fd845d3718f2e2fbaL54 - so I am curious what happens if you pass the object in as a URL (as it was) rather than a String (as it is now)? Do you get an object reference printed out or something?

Reply:
I changed the log parameter for the id from URL to String: https://github.com/jbosstm/narayana/pull/2331/files#diff-e4c877e90cd6988c76f4b37dbd005f96444a75919262742fd845d3718f2e2fbaL54

Reply:
> Actually, I presume it has something to do with how a URL is interpreted in https://github.com/jbosstm/narayana/pull/2331/files#diff-e4c877e90cd6988c76f4b37dbd005f96444a75919262742fd845d3718f2e2fbaL54 - so I am curious what happens if you pass the object in as a URL (as it was) rather than a String (as it is now)? Do you get an object reference printed out or something?

It doesn't compile: `incompatible types: java.net.URL cannot be converted to java.lang.String`